pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -814,6 +815,10 @@ public void connectionOpened(final ClientCnx cnx) {
         if (si != null && (SchemaType.BYTES == si.getType() || SchemaType.NONE == si.getType())) {
             // don't set schema for Schema.BYTES
             si = null;
+        } else {
+            if (schema instanceof AutoConsumeSchema) {
+                si = ((AutoConsumeSchema) schema).getAutoConsumeSchemaInfo();
```

Comment:
Instead of adding a `getAutoConsumerSchemaInfo` method, I think we can just create the `SchemaInfo` here.

```java
            si = SchemaInfo.builder()
                    .name("AutoConsume")
                    .type(SchemaType.AUTO_CONSUME)
                    .schema(new byte[0])
                    .build();
```